query_integration_test.go

```diff
@@ -0,0 +1,85 @@
+package sqlds
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data/sqlutil"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type testArgs struct {
+	MySQLURL            string
+	RunIntegrationTests bool
+}
+
+func testEnvArgs(t *testing.T) testArgs {
+	t.Helper()
+	var args testArgs
+	if val, ok := os.LookupEnv("MYSQL_URL"); ok {
+		args.MySQLURL = val
+	} else {
+		args.MySQLURL = "mysql:mysql@/mysql"
+	}
+
+	if _, ok := os.LookupEnv("INTEGRATION_TESTS"); ok {
+		args.RunIntegrationTests = true
+	}
+
+	return args
+}
+
+func TestQuery_MySQL(t *testing.T) {
+	var (
+		args = testEnvArgs(t)
+		ctx  = context.Background()
+
+		db *sql.DB
+	)
+
+	if !args.RunIntegrationTests {
+		t.SkipNow()
+	}
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	limit := 10
+	for i := 0; i < limit; i++ {
+		d, err := sql.Open("mysql", args.MySQLURL)
+		if err == nil {
+			db = d
+			break
+		}
+
+		<-ticker.C
+	}
+	defer db.Close()
```

Comment:
can you add a comment why the `for` loop and the ticker are needed?

Reply:
done